src/components/AddFavorites/index.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { ReactComponent as StarIcon } from "../../assets/star.svg";
+import { IAddFavoritesProps } from "../../types/interfaces";
+export default function AddFavorites({
+  backgroundColor,
+  textColor,
+  favorited,
+  toggleFavorite
+}: IAddFavoritesProps) {
+  return (
+    <button className="favorites--button" onClick={toggleFavorite}>
+      <div
+        className="favorites--container"
+        style={{ backgroundColor: `var(${backgroundColor})` }}
+      >
+        <StarIcon
+          style={
+            favorited
+              ? {
+                  fill: `var(${textColor})`,
+                  stroke: `var(${textColor})`
+                }
+              : {
+                  stroke: `var(${textColor})`
+                }
+          }
+        />
+        <div className="favorites--text" style={{ color: `var(${textColor})` }}>
+          Add to Favorites
+        </div>
+      </div>
```

Comment:
A button can't hold block components in it. Change your button tag to an `a` tag or `div` tag. And don't forget to add aria-label and role tags for A11Y. You can read more about [accessibility](https://developer.mozilla.org/en-US/docs/Web/Accessibility)

Reply:
I will study it and make my components according to those rules thanks!